webui/react/src/utils/data.ts

```diff
@@ -13,6 +13,8 @@ export const isFiniteNumber = (data: unknown): data is number => isNumber(data)
 export const isObject = (data: unknown): boolean => {
   return typeof data === 'object' && !Array.isArray(data) && !isSet(data) && data !== null;
 };
+export const isJsonObject = (json: Json): json is JsonObject => isObject(json);
```

Comment:
sorry I missed this on my first sweep -- `isObject` will return `true` if `data` is a `Map` or other class -- should we instead check like
```suggestion
export const isJsonObject = (json: Json): json is JsonObject => typeof json === 'object' && Object.getPrototypeOf(json) === Object.prototype;
```
?

Reply:
While that's true those classes cannot be passed to `isJsonObject` because it requires that the input is `Json`.

Reply:
got it!

Reply:
```console
 ERROR(TypeScript)  Argument of type 'Map<string, Json>' is not assignable to parameter of type 'Json'.
  Type 'Map<string, Json>' is not assignable to type 'JsonObject'.
    Index signature for type 'string' is missing in type 'Map<string, Json>'.
 FILE  /Users/claire/Programming/determined/webui/react/src/utils/data.ts:19:14

    18 |
  > 19 | isJsonObject(new Map<string, Json>())
       |              ^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
The only valid inhabitants of `Json` that pass `isObject` are also inhabitants of `JsonObject`